From ef5bfcf668e6029faa78534dfeb2591df854cdef Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 17 Sep 2013 17:28:35 -0500
Subject: [PATCH 161/169] backlight,random-seed: move state files into
 /var/lib/systemd

Let's not scatter (private) files in /var around, let's place them all
in /var/lib/systemd and below.
---
 Makefile.am               | 4 ++--
 TODO                      | 2 --
 src/backlight/backlight.c | 4 ++--
 3 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 9b06c28..8d70ad3 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -159,7 +159,7 @@ AM_CPPFLAGS = \
 	-DSYSTEMD_TTY_ASK_PASSWORD_AGENT_BINARY_PATH=\"$(rootbindir)/systemd-tty-ask-password-agent\" \
 	-DSYSTEMD_STDIO_BRIDGE_BINARY_PATH=\"$(bindir)/systemd-stdio-bridge\" \
 	-DROOTPREFIX=\"$(rootprefix)\" \
-	-DRANDOM_SEED=\"$(localstatedir)/lib/random-seed\" \
+	-DRANDOM_SEED=\"$(localstatedir)/lib/systemd/random-seed\" \
 	-DSYSTEMD_CRYPTSETUP_PATH=\"$(rootlibexecdir)/systemd-cryptsetup\" \
 	-DSYSTEM_GENERATOR_PATH=\"$(systemgeneratordir)\" \
 	-DUSER_GENERATOR_PATH=\"$(usergeneratordir)\" \
@@ -4213,7 +4213,7 @@ substitutions = \
        '|PACKAGE_VERSION=$(PACKAGE_VERSION)|' \
        '|PACKAGE_NAME=$(PACKAGE_NAME)|' \
        '|PACKAGE_URL=$(PACKAGE_URL)|' \
-       '|RANDOM_SEED=$(localstatedir)/lib/random-seed|' \
+       '|RANDOM_SEED=$(localstatedir)/lib/systemd/random-seed|' \
        '|prefix=$(prefix)|' \
        '|exec_prefix=$(exec_prefix)|' \
        '|libdir=$(libdir)|' \
diff --git a/TODO b/TODO
index d5b32ac..c0f51de 100644
--- a/TODO
+++ b/TODO
@@ -72,8 +72,6 @@ Features:
   interate through that to bring down complexity from O(n^2) to O(n)
   when loading units
 
-* Move backlight and random-seed into /var/lib/systemd
-
 * If we try to find a unit via a dangling symlink generate a clean
   error. Currently we just ignore it and read the unit from the search
   path anyway.
diff --git a/src/backlight/backlight.c b/src/backlight/backlight.c
index 1ef0b45..9b2eada 100644
--- a/src/backlight/backlight.c
+++ b/src/backlight/backlight.c
@@ -42,7 +42,7 @@ int main(int argc, char *argv[]) {
 
         umask(0022);
 
-        r = mkdir_p("/var/lib/backlight", 0755);
+        r = mkdir_p("/var/lib/systemd/backlight", 0755);
         if (r < 0) {
                 log_error("Failed to create backlight directory: %s", strerror(-r));
                 goto finish;
@@ -72,7 +72,7 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
-        saved = strappend("/var/lib/backlight/", udev_device_get_sysname(device));
+        saved = strappend("/var/lib/systemd/backlight/", udev_device_get_sysname(device));
         if (!saved) {
                 r = log_oom();
                 goto finish;
-- 
1.8.4

